src/Interpreters/SystemLog.h

```diff
@@ -207,11 +208,13 @@ SystemLog<LogElement>::SystemLog(Context & context_,
     const String & storage_def_,
     size_t flush_interval_milliseconds_)
     : context(context_)
+    , insert_context(Context(context_))
     , table_id(database_name_, table_name_)
     , storage_def(storage_def_)
     , flush_interval_milliseconds(flush_interval_milliseconds_)
 {
     assert(database_name_ == DatabaseCatalog::SYSTEM_DATABASE);
+    insert_context.makeQueryContext(); // we need query context to do inserts to target table with MV containing subqueries or joins
```

Comment:
I'm not sure if it's ok to store query context for a whole object lifetime, it may have some negative side effects see
https://github.com/ClickHouse/ClickHouse/commit/7e7fe3c441ffeaff8e14ac3419d90b40a4f73b94

Maybe it would be safer to create a new Context for every flush? Would it be too expensive?

Reply:
Probably yes, it also may have negative effects if [structure of some tables have been changed](https://github.com/ClickHouse/ClickHouse/blob/45adacf0bc0ff5db21e5b711a99202a87837027e/src/Interpreters/InterpreterSelectWithUnionQuery.cpp#L166). Context copying is not that expensive (at least we do it multiple times for each query), so I think it's ok to create it on every flush (once a 7 seconds for each `SystemLog` table).

Reply:
changed